tests/lib/rules/sort-character-class-elements.ts

```diff
@@ -230,5 +233,57 @@ tester.run("sort-character-class-elements", rule as any, {
                 "Expected character class elements to be in ascending order. '\t' should be before ' '.",
             ],
         },
+        {
+            code: String.raw`/[a[a]\q{a}[a--b]]/v`,
+            output: String.raw`/[[a]a\q{a}[a--b]]/v`,
+            errors: [
+                "Expected character class elements to be in ascending order. '[a]' should be before 'a'.",
+                "Expected character class elements to be in ascending order. '\\q{a}' should be before 'a'.",
+            ],
+        },
+        {
+            code: String.raw`/[[a]a\q{a}[a--b]]/v`,
+            output: String.raw`/[\q{a}[a]a[a--b]]/v`,
+            errors: [
+                "Expected character class elements to be in ascending order. '\\q{a}' should be before '[a]'.",
+            ],
+        },
+        {
+            code: String.raw`/[\q{a}[a]a[a--b]]/v`,
+            output: String.raw`/[\q{a}[a][a--b]a]/v`,
```

Comment:
Is this really the order we should go for? I would have expected the order to be `[a\q{a}[a][a--b]]`.

I think that the order `*` > `\q` > `[]` might be better, because it prevents cases like `[\q{a}[a][a--b]a]`. That tiny last `a` is easy to miss, but that's not a problem with `[a\q{a}[a][a--b]]` IMO.

Thoughts?

Reply:
Thank you for your thoughts!
I think the order you suggested is easier to read. I changed the default order.